src/components/tools/table-tool/cell-formatter.tsx

```diff
@@ -1,14 +1,33 @@
-import { TFormatterProps } from "./table-types";
+import React from "react";
+import { kCellHorizontalPadding, kCellLineHeight, kDefaultColumnWidth, kRowHeight, TRow } from "./table-types";
 import { useNumberFormat } from "./use-number-format";
 
-export const formatValue = (formatter: (n: number | { valueOf(): number }) => string, value: any) => {
-  if ((value == null) || (value === "")) return "";
+import './cell-formatter.scss';
+
+export const formatValue = (
+    formatter: (n: number | { valueOf(): number }) => string, value: any,
+    width?: number, row?: TRow, rowHeight?: (args: any) => number
+  ) => {
+  if ((value == null) || (value === "")) return <span></span>;
   const num = Number(value);
-  if (!isFinite(num)) return value;
-  return formatter(num);
+  if (!isFinite(num)) {
+    const cellWidth = (width || kDefaultColumnWidth) - kCellHorizontalPadding;
+    const height = rowHeight && row ? rowHeight({ row }) : kRowHeight;
```

Comment:
Calling `rowHeight()` here is going to recompute the row height as part of the rendering of every cell. Probably not worth addressing right now but a couple of alternatives are mentioned in a separate comment.